packages/block-library/src/html/edit.js

```diff
@@ -42,6 +46,14 @@ export default function HTMLEdit( { attributes, setAttributes, isSelected } ) {
 		];
 	}, [] );
 
+	useEffect( () => {
+		if ( !! showPreview ) {
```

Comment:
Thank you @mauriac for working on this. I'm thinking do we really need  `if-else` condition? While doing review I observed both the variables `isPreview` and `showPreview ` are in sync if we modify current statement something like this:

```diff
-const [ isPreview, setIsPreview ] = useState();
+const [ isPreview, setIsPreview ] = useState(false);
```

PS: Current default value of `isPreview` variable is `undefined`.

Looking forward to hearing from you

Reply:
Hello @amustaque97 thank you for your review, but I am a bit lost;

I do not get the suggestion. Please do you mind enlighten me?

The `useEffect` statement is just a plus, but on second thought let's remove it.

Thank you.

Reply:
Hello @amustaque97 thank you for your review, but I am a bit lost;

I do not get the suggestion. Please do you mind enlighten me?

The sync between isPreview and showPreview is just a plus, but on second thought let's remove it.

Thank you.

Reply:
> The sync between isPreview and showPreview is just a plus, but on second thought let's remove it.

What I mean to say is that we can remove if-else block and can write something like this 
``` javascript
useEffect(() => {
    setIsPreview( showPreview );
}, [showPreview])
```



Also, to change line [here ](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/html/edit.js#L22)
```diff
-const [ isPreview, setIsPreview ] = useState();
+const [ isPreview, setIsPreview ] = useState(false);
```


I hope I'm able to explain request changes.

Thank you.

Reply:
@mauriac , I saw your changes. I will take a proper look at it in sometime and update my comments here.

Reply:
> > The sync between isPreview and showPreview is just a plus, but on second thought let's remove it.
> 
> What I mean to say is that we can remove if-else block and can write something like this
> 
> ```js
> useEffect(() => {
>     setIsPreview( showPreview );
> }, [showPreview])
> ```
>
> Also, to change line [here ](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/html/edit.js#L22)
> 
> ```diff
> -const [ isPreview, setIsPreview ] = useState();
> +const [ isPreview, setIsPreview ] = useState(false);
> ```
> 
> I hope I'm able to explain request changes.
> 
> Thank you.

Oh yes you are totally right.  I got it now, I make some changes, give me your back when you have sometime. thank you.

Reply:
Your changes makes more sense to load preview setting once. Also, can we change `useEffect` block to look like this:

```javascript
useEffect( () => {
    if ( showPreview ) {
        switchToPreview();
    }
}, [] );
```
I think PR is good to approve after the above final changes. Let me know what you think?

Reply:
> Your changes makes more sense to load preview setting once. Also, can we change `useEffect` block to look like this:
> 
> ```js
> useEffect( () => {
>     if ( showPreview ) {
>         switchToPreview();
>     }
> }, [] );
> ```
> 
> I think PR is good to approve after the above final changes. Let me know what you think?

I agree, I applied changes, so what about this change
`+const [ isPreview, setIsPreview ] = useState(false);` should we still apply it?